### PR TITLE
Add GA4 to pocket pages

### PR DESCRIPTION
**Pull request by nathan-barrett**
## One-line summary
Adds GA4 to the pocket pages

## Issue / Bugzilla link
#12432 

## Testing
To test this work:

- [ ] When you load any of the pocket pages, google tag manager should be loaded

### CODE REVIEW

**Comment by nathan-barrett** [comment]
@alexgibson Not sure if there is a better way about adding GA4 but, the inline script is being blocked by our CSP (from the `pocket/includes/google-analytics.html` file). Whats the best move here?

File: bedrock/pocket/templates/pocket/base.html [hunk 1]
```diff
@@ -35,6 +35,9 @@
         function OptanonWrapper() {}
```
[the remark this answers is not in the document]
Comment (maureenlholland) [reply bedrock/pocket/templates/pocket/base.html:35]: ah ok, that's the UA ID, we might want to create a settings variable for Pocket GTM ID and update the comments here to reflect that Pocket will be running both analytics scripts for a while going forward: https://github.com/mozilla/bedrock/commit/a051d005d91b251f5bf6655661b2ba1f5537e5a1

[the remark this answers is not in the document]
Comment (nathan-barrett) [reply bedrock/pocket/templates/pocket/base.html:35]: WHen testing this you'll need to add this to your `.env` file: `GOOGLE_ANALYTICS_GTM_ID="G-NFR9Y40GD3"`

**Comment by nathan-barrett** [comment]
@maureenlholland - bumping this - wasn't sure if you needed to take another look at this or if there was more work on my end

**Comment by maureenlholland** [comment]
🤦‍♀️ thanks for the ping, totally slipped off my radar. Will take a closer look after web dev meeting.

**Review by maureenlholland — APPROVED** [review]
r+ LGTM

one question for yourself or a backend dev: does the GTM ID live in a settings file somewhere for production? works with local .env and looks like pocket-demo is set too, but I don't know where production would get the GTM value

**Comment by nathan-barrett** [comment]
@stevejalim is there anyplace that we'd need to add the GTM Container ID for prod? (see maureen's comment)

File: bedrock/settings/base.py [hunk 2]
```diff
@@ -1183,7 +1183,9 @@ def get_app_name(hostname):
 
 GA_ACCOUNT_CODE = ""  # DELETE ME: Deprecated?
 GTM_CONTAINER_ID = config("GTM_CONTAINER_ID", default="")  # NB: Not used in all Bedrock modes (Mozorg only).
+# Pocket mode will be running both GA UA and GA4 for a while going forward
 GOOGLE_ANALYTICS_ID = config("GOOGLE_ANALYTICS_ID", default="")  # NB: Not used in all Bedrock modes (Pocket only).
+GOOGLE_ANALYTICS_GTM_ID = config("GOOGLE_ANALYTICS_GTM_ID", default="")  # Not used in all Bedrock modes (Pocket only).
```
Comment (stevejalim) [inline bedrock/settings/base.py:1188]: This looks like it's a second env var for loading the GTM ID. If I'm wrong and both things aren't for Google Tag Manager, ignore me. But if I'm right I suggest instead you re-use `GTM_CONTAINER_ID` for Pocket mode too (adjusting comments appropriately), then we have just one GA slot and one GTM slot.

Either way, in terms of setting a GTM ID for Pocket mode, that needs to happen in www-config: add it just above here in www-config https://github.com/mozmeao/www-config/blob/main/iowa-a/bedrock-prod/pocket-deploy.yaml#L116 (that's prod, but you may also want to add it for dev and stage too, maybe?)

File: gcp/bedrock-demos/cloudrun/pocket-demo.env.yaml [hunk 3]
```diff
@@ -13,6 +13,7 @@ DB_DOWNLOAD_IGNORE_GIT: "True"
 DEBUG: "False"
 DEV: "True"
 GOOGLE_ANALYTICS_ID: "UA-370613-9"
+GOOGLE_ANALYTICS_GTM_ID: "G-NFR9Y40GD3"
```
Comment (stevejalim) [inline gcp/bedrock-demos/cloudrun/pocket-demo.env.yaml:16]:
```suggestion
GTM_CONTAINER_ID: "G-NFR9Y40GD3"
```

File: bedrock/pocket/templates/pocket/includes/google-analytics.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{#
+ This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#}
+
+{% if settings.GOOGLE_ANALYTICS_GTM_ID %}
+<!-- Google tag (gtag.js) -->
+<script async src="https://www.googletagmanager.com/gtag/js?id={{ settings.GOOGLE_ANALYTICS_GTM_ID }}"></script>
```
Comment (stevejalim) [inline bedrock/pocket/templates/pocket/includes/google-analytics.html:9]:
```suggestion
<script async src="https://www.googletagmanager.com/gtag/js?id={{ settings.GTM_CONTAINER_ID }}"></script>
```

File: bedrock/settings/base.py (hunk 2 above)
Comment (nathan-barrett) [reply bedrock/settings/base.py:1188]: makes sense, thanks @stevejalim

**Comment by nathan-barrett** [comment]
Created this PR in www-config: https://github.com/mozmeao/www-config/pull/500 to add the GTM_CONTAINER_ID for pocket deploys

**Comment by nathan-barrett** [comment]
Just following up on this from last year - does this look good to merge @stevejalim @maureenlholland?

**Review by stevejalim — APPROVED** [review]
Code LGTM - I haven't test driven, but if @maureenlholland has then it's a 👍

**Review by maureenlholland — APPROVED** [review]
r++ with the www-config settings PR, nice work ✅

**Pull request merged**